test/test_socket_options.cpp

```diff
@@ -353,26 +524,77 @@ bool CheckInvalidValues(const OptionTestEntry& entry, SRTSOCKET sock, const char
     return true;
 }
 
-TEST_F(TestSocketOptions, DefaultVals)
+void TestDefaultValues(SRTSOCKET s)
 {
+    const char* test_desc = "[Caller, default]";
     for (const auto& entry : g_test_matrix_options)
     {
-        const char* test_desc = "[Caller, default]";
+        // Check flags. An option must be RW to test default value
+        const bool is_group = (s & SRTGROUP_MASK) != 0;
+
+        if (!(entry.flags & Flags::R))
+        {
+            // TODO: Check reading retuns an error.
+            LOGD(cerr << "Skipping " << entry.optname << ": not readable.\n");
+            continue; // The flag must be READABLE and WRITABLE for this.
+        }
+
+        // Check that retrieving a value must fail if the option is not a group option read on a group and not a socket
+        // option read on a socket.
+        bool readable = true;
+        if (is_group)
+        {
+            readable = entry.allof(Flags::G) && entry.anyof(Flags::I, Flags::D);
```

Comment:
Same as?
```suggestion
            readable = entry.allof(Flags::G, Flags::I, Flags::D);
```

Reply:
No, this is wrong. Flags I and D don't come together. What you wrote here is all of G, I and D must be set. We need G and either I or D.

Reply:
Ah, right, sorry, I missed that different functions are used: "all" and "any"